src/Owin.Security.OpenIdConnect.Server/OpenIdConnectServerHandler.Endpoints.cs

```diff
@@ -1770,6 +1771,81 @@ internal partial class OpenIdConnectServerHandler : AuthenticationHandler<OpenId
             if (notification.HandledResponse) {
                 return;
             }
+            
+            var identifier = ticket.GetProperty(OpenIdConnectConstants.Extra.ClientId);
```

Comment:
Another remark: all these checks come far too late. Ideally, they shouldn't be called after the `ValidationEndpoint` event. You should move this whole block at line 1732.